index.html

```diff
@@ -1974,5 +1974,124 @@ <h2>MediaStream in workers</h2>
+      from remote participants (i.e., the teacher) to be cancelled in order to
+      avoid echo, but also requires that the accompaniment not be cancelled
+      since the music teacher on the remote side needs to hear it together with
+      the sound from the instrument.
+      To support these use cases, we introduce the {{echoCancellationMode}}
+      constrainable property.
+    </p>
+    <p>The WebIDL changes are the following:</p>
+    <pre class="idl">
+enum EchoCancellationModeEnum {
+  "private",
+  "remote-only",
+  "flexible",
+  "none"
+};
+    </pre>
+    <table data-link-for="EchoCancellationModeEnum" data-dfn-for="EchoCancellationModeEnum" class="simple">
+      <caption><dfn>EchoCancellationModeEnum</dfn> Enumeration description</caption>
+      <thead>
+        <tr>
+          <th>Enum value</th><th>Description</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td><dfn><code id=
+          "idl-def-EchoCancellationModeEnum.private">private</code></dfn></td>
+          <td>
+            <p>The system tries to cancel all local audio sources from the
+              microphone signal. This is compatible only with an
+              {{MediaTrackConstraintSet/echoCancellation}} value of <code>true</code>.</p>
+          </td>
+        </tr>
+        <tr>
+          <td><dfn><code id=
+          "idl-def-EchoCancellationModeEnum.remote-only">remote-only</code></dfn></td>
+          <td>
+            <p>The system tries to cancel from the microphone signal audio
+              received from RTCPeerConnections. This is compatible only with an
+              {{MediaTrackConstraintSet/echoCancellation}} value of <code>true</code>.</p>
+          </td>
+        </tr>
+        <tr>
+          <td><dfn><code id=
+          "idl-def-EchoCancellationModeEnum.left">flexible</code></dfn></td>
+          <td>
+            <p>The user agent decides what audio to cancel from the microphone
+              signal. It should cancel at least audio received from
+              RTCPeerConnections. This is compatible only with an
+              {{MediaTrackConstraintSet/echoCancellation}} value of <code>true</code>.</p>
+          </td>
+        </tr>
+        <tr>
+          <td><dfn><code id=
+          "idl-def-EchoCancellationModeEnum.none">none</code></dfn></td>
+          <td>
+            <p>No audio is cancelled from the microphone signal. This is
+              compatible only with an {{MediaTrackConstraintSet/echoCancellation}} value of
+              <code>false</code>.</p>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+    <pre class="idl">
+partial dictionary MediaTrackSupportedConstraints {
+  boolean echoCancellationMode = true;
+};
+
+partial dictionary MediaTrackCapabilities {
+  sequence&lt;DOMString&gt; echoCancellationMode;
+};
+
+partial dictionary MediaTrackConstraintSet {
+  ConstrainDOMString echoCancellationMode;
+};
```

Comment:
It if's a separate mode then it's unclear how it interacts with the existing constraint. I propose we make it a DOMString || boolean union as discussed in the meeting.

I'm open to doing this i mediacapture-main if it's easier.